### PR TITLE
Download and organise DLRM data automatically 

### PR DESCRIPTION
**Pull request by anandhu-eng**

### CODE REVIEW

File: script/get-dlrm-data-mlperf-inference/customize.py [hunk 1]
```diff
@@ -22,38 +22,48 @@ def preprocess(i):
...
+        if env['CM_DLRM_DATASET_DOWNLOAD'] != True:
+            if not os.path.exists(os.path.join(dlrm_data_path, "criteo", "day23", "fp32", "day_23_sparse_multi_hot_unpacked")):
+                os.system(f"unzip {os.path.join(dlrm_data_path, 'criteo', 'day23', 'fp32', 'day_23_sparse_multi_hot.npz')} -d {os.path.join(dlrm_data_path, 'criteo', 'day23', 'fp32', 'day_23_sparse_multi_hot_unpacked')}")
 
-        if not os.path.exists(os.path.join(dlrm_data_path, "criteo", "day23", "fp32", "day_23_sparse_multi_hot_unpacked")):
-            os.system(f"unzip {os.path.join(dlrm_data_path, 'criteo', 'day23', 'fp32', 'day_23_sparse_multi_hot.npz')} -d {os.path.join(dlrm_data_path, 'criteo', 'day23', 'fp32', 'day_23_sparse_multi_hot_unpacked')}")
+            if os.path.exists(os.path.join(dlrm_data_path, "criteo", "day23", "fp32", "day_23_sparse_multi_hot.npz")):
+                file_path = os.path.join(dlrm_data_path, "criteo", "day23", "fp32", "day_23_sparse_multi_hot.npz")
+                run_cmd = ("echo {} {} | md5sum -c").format('c46b7e31ec6f2f8768fa60bdfc0f6e40', file_path)
```
Comment (arjunsuresh) [inline script/get-dlrm-data-mlperf-inference/customize.py:61]: we want to do md5sum check even for download right?

Comment (anandhu-eng) [reply script/get-dlrm-data-mlperf-inference/customize.py]: Yes... I tried to handle it in commit 19862dd

File: script/get-dlrm-data-mlperf-inference/_cm.yaml [hunk 2]
```diff
@@ -16,6 +16,20 @@ new_env_keys:
   - DLRM_DATA_PATH
 input_mapping:
   dlrm_data_path: CM_DLRM_DATA_PATH
+
+prehook_deps:
+  - tags: get,ml-model,dlrm,_pytorch
+    update_tags_from_env_with_prefix:
+      download_path: CM_DLRM_DATA_PATH
```
Comment (arjunsuresh) [inline script/get-dlrm-data-mlperf-inference/_cm.yaml:23]: `_` missing? Also this uses a list right?

File: script/get-dlrm-data-mlperf-inference/_cm.yaml [hunk 3]
```diff
@@ -16,6 +16,20 @@ new_env_keys:
...
+    enable_if_env:
+      CM_DLRM_MODEL_DOWNLOAD:
+        - "on"
+  - tags: get,dataset,preprocessed,criteo,_mlc
+    update_tags_from_env_with_prefix:
+      output_dir: CM_DLRM_DATA_PATH
```
Comment (arjunsuresh) [inline script/get-dlrm-data-mlperf-inference/_cm.yaml:29]: Is `output_dir` a tag?

File: script/get-dlrm-data-mlperf-inference/customize.py [hunk 4]
```diff
@@ -22,46 +37,61 @@ def preprocess(i):
...
-
-
-        if not os.path.exists(os.path.join(dlrm_data_path, "criteo", "day23", "fp32", "day_23_sparse_multi_hot_unpacked")):
-            os.system(f"unzip {os.path.join(dlrm_data_path, 'criteo', 'day23', 'fp32', 'day_23_sparse_multi_hot.npz')} -d {os.path.join(dlrm_data_path, 'criteo', 'day23', 'fp32', 'day_23_sparse_multi_hot_unpacked')}")
-
-        xsep = ' && '
+            print(f'day_23_labels.npy is missing inside {dlrm_data_path}/criteo/day23/fp32')
+            env['CM_DLRM_DATASET_DOWNLOAD'] = True
+        # TBD: raw data is not available with mlcommons hosted dataset
+        #if not os.path.exists(os.path.join(dlrm_data_path, "criteo", "day23", "raw_data")):
```
Comment (arjunsuresh) [inline script/get-dlrm-data-mlperf-inference/customize.py:64]: We should now turn this check on and ask user to point to this file via an input say `criteo_day23_raw_data_path`. [This](https://criteo.wetransfer.com/downloads/4bbea9b4a54baddea549d71271a38e2c20230428071257/d4f0d2) is the link to download the rawdata but only manual download is possible.

Comment (anandhu-eng) [reply script/get-dlrm-data-mlperf-inference/customize.py:64]: would `criteo_raw_data_path` and `CM_CRITEO_RAW_DATA_PATH` be a good naming option for input mapping and env variable?

Comment (arjunsuresh) [reply script/get-dlrm-data-mlperf-inference/customize.py:64]: We need day23 too as we only need to download day23 file - not the entire raw data.

**Review by arjunsuresh — APPROVED** [review]

File: script/get-dlrm-data-mlperf-inference/customize.py (hunk 4 above)
Comment (anandhu-eng) [reply script/get-dlrm-data-mlperf-inference/customize.py]: [1dec20a](https://github.com/GATEOverflow/cm4mlops/pull/93/commits/1dec20a743cd14b34d3521de3b1ab6637982cfb0) incorporated in this commit

**Pull request merged**